src/replicache-like.ts

```diff
@@ -0,0 +1,45 @@
+import {ReadonlyJSONValue} from './json.js';
+
+type DiffOperationAdd<Key, Value = ReadonlyJSONValue> = {
+  readonly op: 'add';
+  readonly key: Key;
+  readonly newValue: Value;
+};
+type DiffOperationDel<Key, Value = ReadonlyJSONValue> = {
+  readonly op: 'del';
+  readonly key: Key;
+  readonly oldValue: Value;
+};
+type DiffOperationChange<Key, Value = ReadonlyJSONValue> = {
+  readonly op: 'change';
+  readonly key: Key;
+  readonly oldValue: Value;
+  readonly newValue: Value;
+};
+type DiffOperation<Key> =
+  | DiffOperationAdd<Key>
+  | DiffOperationDel<Key>
+  | DiffOperationChange<Key>;
+type NoIndexDiff = readonly DiffOperation<string>[];
```

Comment:
Since Rails doesn’t currently do anything with indexes, this can be a little smaller.

Reply:
Done in https://github.com/rocicorp/rails/pull/41